src/plugins/borrow-plugins/types.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import { type EdgeAccount, type EdgeCurrencyWallet, type EdgeNetworkFee } from 'edge-core-js'
+
+// Proposal for a "service plugin" which is a Higher-Order-Plugin that can be
+// instantiated with some account context and returns a service instance.
+export type EdgeServicePlugin<Info, Api> = {
+  info: Info,
+  init: (account: EdgeAccount) => Promise<Api>
+}
```

Comment:
Borrow plugins do not get access to the account. This is a hold-over from back when we had the action plugin idea confused with the borrow plugin idea. So, we can just get rid of this indirection, and put `BorrowPluginInfo` right on `BorrowService`. So basically, `BorrowService` gets renamed to `BorrowPlugin`.

Reply:
Without access to `EdgeAccount`, how will the borrow plugin determine which wallets have "loan accounts"?

Reply:
Resolution: Remove `EdgeServicePlugin` in favor of an explicit `BorrowPlugin` which has a `makeBorrowEngine` method which expects a `EdgeCurrencyWallet` and returns a `BorrowEngine` object which is a rename for `BorrowService`.